runtime/native.go

```diff
@@ -348,6 +369,19 @@ func getNativeType(rtype reflect.Type) *Type {
 	return t
 }
 
+func newNativeField(name string, i, refDepth int) *Object {
+	// Closes over `i` and `refDepth`
+	nativeFieldGet := func(f *Frame, desc, instance *Object, owner *Type) (*Object, *BaseException) {
+		v := toNativeUnsafe(instance).value
+		for i := 0; i < refDepth; i++ {
+			v = v.Elem()
+		}
+		return WrapNative(f, v.Field(i))
+	}
+
+	return &Object{typ: &Type{slots: typeSlots{Get: &getSlot{nativeFieldGet}}}}
```

Comment:
I think you can use the existing `property` type for this:

```
return newProperty(
  newBuiltinFunction(fieldName, func(f *Frame, args Args, _ KWargs) (*Object, *BaseException) {
    if raised := checkFunctionArgs(f, fieldName, args, theNativeType); raised != nil {
      return nil, raised
    }
    v := toNativeUnsafe(args[0]).value
  }).ToObject(), nil, nil)
)
```

Note that I'm validating that arg[0] has the expected Python type before calling toNativeUnsafe(). You must only use the to*Unsafe() functions when you're sure that the argument is an instance of that type. Otherwise you're doing an unsafe cast between incompatible types (e.g. Str and Float).